kpops/components/base_components/kafka_app.py

```diff
@@ -131,6 +131,14 @@ class KafkaApp(PipelineComponent, ABC):
         description=describe_attr("app", __doc__),
     )
 
+    @pydantic.model_validator(mode="after")
+    def warning_for_latest_image_tag(self) -> Self:
+        if self.app.image_tag == "latest" and "$" not in self.name:
```

Comment:
The validation was called before `self.name` got substituted, which is why this condition exists.